distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -1145,6 +1148,126 @@ type Table
             Table.Table_Data (Java_Table.concat java_tables.to_array)
         Table.Table_Data other_java_table -> Table.Table_Data (Java_Table.concat [self.java_table, other_java_table].to_array)
 
+    ## Returns a new table with a chosen subset of columns left unchanged and
+       the other columns pivoted to rows with a single name field and a single
+       value field.
+
+       Arguments:
+       - id_fields: Set of fields to remain as columns. These values will be
+         repeated for each data field that is pivoted.
+       - name_field: The name of the field that will contain the names of the
+         pivoted fields. If this name is already in use, it will be renamed
+         with a numeric suffix.
+       - value_field: The name of the field that will contain the values of the
+         pivoted fields. If this name is already in use, it will be renamed
+         with a numeric suffix.
+       - on_problems: Specifies how to handle problems if they occur, reporting
+         them as warnings by default.
+
+         The following problems can occur:
+         - If a column in columns is not in the input table, a
+           `Missing_Input_Columns`.
+         - If duplicate columns, names or indices are provided, a
+           `Duplicate_Column_Selectors`.
+         - If a column index is out of range, a `Column_Indexes_Out_Of_Range`.
+         - If two distinct indices refer to the same column, an
+           `Input_Indices_Already_Matched`, with the column included the first
+           time it is matched.
+         - If there are no columns to transpose, a `No_Input_Columns`.
+    transpose : Vector Text | Column_Selector -> Text -> Text -> Problem_Behavior -> Table
+    transpose self (id_fields = Column_Selector.By_Name []) (name_field="Name") (value_field="Value") (on_problems = Report_Warning) =
+        columns_helper = self.columns_helper
+        unique = Unique_Name_Strategy.new
+        problem_builder = Problem_Builder.new
+
+        id_columns = columns_helper.select_columns_helper id_fields False problem_builder
+        count = self.column_count - id_columns.length
+        if count == 0 then problem_builder.report_other_warning No_Input_Columns_Selected
+
+        selected_names = Map.from_vector (id_columns.map column-> [column.name, True])
+
+        data = columns_helper.internal_columns.filter column->(selected_names.get_or_else column.name False . not)
+        java_data = data.map .java_column
+
+        java_id = id_columns.map .java_column
+
+        id_columns.each k-> unique.make_unique k.name
+        result = Table.Table_Data (Java_Table.transpose java_id.to_array java_data.to_array (unique.make_unique name_field) (unique.make_unique value_field))
```

Comment:
```suggestion
        result = Table.Table_Data (Java_Table.transpose java_id java_data (unique.make_unique name_field) (unique.make_unique value_field))
```

I think with the Vector being a builtin and implementing the Truffle hasArrayElements API, we do not need the `to_array` calls. At least I've been _not_ using them in some new code and it was all working OK. Unless you are using them for some specific reason that I did not consider? If yes please let me know, as I probably should adapt my usages too, as we should be consistent.

Reply:
Right! I was using `List<>` in my case indeed. Well, then we can also keep it as-is, seems fine.